src/Examine/LuceneEngine/DeletePolicies/NoDeletionPolicy.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System.Collections.Generic;
+using Lucene.Net.Index;
+
+namespace Examine.LuceneEngine.DeletePolicies
+{
+    public class NoDeletionPolicy : IndexDeletionPolicy
```

Comment:
@Shazwazza that one will actually make removal of files, so it will cause End of file exception. so sadly we need deal with that in that way, as we dont make backups, we actively using indexes.
but for 4.8 I will look into Replicator as it is actually more or less what I am doing now :)

Reply:
I had a read of that document along with https://cwiki.apache.org/confluence/display/solr/SolrReplication. Using the snapshotdeletionpolicy sounded like the way to go to me. It allows for hot backups. It wouldn't be a duplex R/W approach, but use the on commit method to choose how many versions to hold and send to blob storage. At the same time you could queue a message to signal the other instances to sync index files. This way you would not need to check the storage account files every query.

Reply:
@nzdev you forgot about one thing, solr is using that internally, we have delay between push and pull, so that will cause EOF :) sadly that aproach would be perfect for doing syncTemp or backups, but not live synchronization :)

Reply:
I think it could be made to work as the SnapshotDeletionPolicy can prevent files from being deleted, so no risk of EOF. Each IndexCommit tracks the changed files. We could then write the files plus the list of changes to blob storage and have the replicas/ client either poll for changes or check on index reader creation. The  looks like the way to go though for 4.8, but ultimately it will be using the same approach as solr. Have a read of the how does it work from https://cwiki.apache.org/confluence/display/solr/SolrReplication. The delay between push and pull can be handled by versioning segments.gen in blob storage.

Reply:
@nzdev hmmm i did read that, but yeah it will be something what will need further investigation, but yeah so far i have working production site base on code which is used in that repo already:
https://github.com/bielu/Examine.Directories
so if we will make changes there I guess we will need do modification of remote directory and test again :)

Reply:
@nzdev @Shazwazza I looked into that more deeply, and there is few limitation:

- all writers has to be closed when restoring (so restoring from blob storage) 
- based on point 1 we still staying with shuffle of folders in time of restore (so event on lucene index still needed)
- hmmm with nodeletionpolicy I hit interesting issue (some documents are still present even if in the newest commit they are not existing) 
I started looking into snapshot policy as it would make much faster restore as it would be only duplication of last commit, but from that what I am seeing we are stacking with same issue, but will investigate this weekend further as might be to that better solution